controller/instance_manager_controller.go

```diff
@@ -1702,34 +1701,52 @@ func (imc *InstanceManagerController) deleteOrphans(im *longhorn.InstanceManager
 	return nil
 }
 
-func (imc *InstanceManagerController) isOrphanDeletable(orphan *longhorn.Orphan, autoDelete bool, instanceMap instanceProcessMap, isInstanceScheduledOnNode func(node, instance string) (bool, error)) (isDeletable bool, err error) {
-	if autoDelete {
-		return true, nil
+func (imc *InstanceManagerController) deleteOrphan(orphan *longhorn.Orphan, resourceDeletable bool) error {
+	if orphan.Spec.ResourceDeletable != resourceDeletable {
+		orphan.Spec.ResourceDeletable = resourceDeletable
+		_, err := imc.ds.UpdateOrphan(orphan)
+		switch {
+		case datastore.ErrorIsNotFound(err):
+			return nil
+		case err != nil:
+			return errors.Wrapf(err, "failed to update resource deletable on orphan %v before deletion", orphan.Name)
+		}
 	}
```

Comment:
Suggest moving the update logic into a separate method so the method name reflects its purpose.